src/software/ai/hl/stp/tactic/dribble/simulated_dribble_tactic_test.cpp

```diff
@@ -0,0 +1,291 @@
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+TEST_F(SimulatedMoveTacticTest, test_ball_bounce_of_enemy_robot)
+{
+    Point initial_position = Point(-3, 1.5);
+    setBallState(BallState(Point(0, 0), Vector(2.5, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(3, 3), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), Point(1, 2.5), Point(1, -2.5), field().enemyGoalCenter(),
+         field().enemyDefenseArea().negXNegYCorner(),
+         field().enemyDefenseArea().negXPosYCorner()}));
+    setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::FORCE_START);
+
+    auto tactic = std::make_shared<DribbleTactic>();
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, tactic](std::shared_ptr<World> world_ptr,
+                       ValidationCoroutine::push_type& yield) {
+            checkPossession(tactic, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+TEST_F(SimulatedMoveTacticTest, test_moving_ball_dribble_dest)
+{
+    Point initial_position    = Point(-3, 1.5);
+    Point dribble_destination = Point(-3, 1);
+    setBallState(BallState(Point(3, -2), Vector(-1, 2)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 2.5), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), Point(1, 2.5), Point(1, -2.5), field().enemyGoalCenter(),
+         field().enemyDefenseArea().negXNegYCorner(),
+         field().enemyDefenseArea().negXPosYCorner()}));
+    setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::FORCE_START);
+
+    auto tactic = std::make_shared<DribbleTactic>();
+    tactic->updateControlParams(dribble_destination, std::nullopt);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, dribble_destination, tactic](std::shared_ptr<World> world_ptr,
+                                            ValidationCoroutine::push_type& yield) {
+            checkPossession(tactic, world_ptr, yield);
+            ballAtPoint(dribble_destination, world_ptr, yield);
+            checkPossession(tactic, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+TEST_F(SimulatedMoveTacticTest, test_moving_ball_dribble_dest_and_orientation)
```

Comment:
In addition to checking that the ball ends up in the desired destination, do we need a validation function that checks that the robot in possession of the ball has the correct orientation?

Reply:
the functions are being added in https://github.com/UBC-Thunderbots/Software/pull/1996. I'll add them once that's merged in

Reply:
added